src/containers/Group/Group.test.tsx

```diff
@@ -1,33 +1,45 @@
 import React from 'react';
-import { shallow } from 'enzyme';
-import { Group } from './Group';
-import { GET_GROUPS } from '../../graphql/queries/Group';
+import { render, screen, wait } from '@testing-library/react';
+import UserEvent from '@testing-library/user-event';
 import { MockedProvider } from '@apollo/client/testing';
 
+import { Group } from './Group';
+import { getGroupQuery, getGroupsQuery, getGroupUsersQuery } from '../../mocks/Group';
+import { getUsersQuery } from '../../mocks/User';
+import { getOrganizationQuery } from '../../mocks/Organization';
+
 const mocks = [
-  {
-    request: {
-      query: GET_GROUPS,
-    },
-    result: {
-      data: [
-        {
-          groups: {
-            label: 'Staff group',
-            description: 'Only for staff members',
-          },
-        },
-      ],
-    },
-  },
+  getUsersQuery,
+  ...getOrganizationQuery,
+  getGroupQuery,
+  getGroupQuery, // if you refetch then you need to include same mock twice, learnt it hard way
+  getGroupUsersQuery,
+  getGroupsQuery,
 ];
 
-const wrapper = shallow(
-  <MockedProvider mocks={mocks}>
+const wrapper = (
+  <MockedProvider mocks={mocks} addTypename={false}>
     <Group match={{ params: { id: 1 } }} />
   </MockedProvider>
 );
 
-it('should render Group', () => {
-  expect(wrapper.exists()).toBe(true);
+describe('<Group />', () => {
+  test('should render Group and hit save', async () => {
+    const { getByText, getAllByTestId } = render(wrapper);
+
+    // loading is show initially
+    expect(getByText('Loading...')).toBeInTheDocument();
+
+    await wait();
+    expect(getByText('Edit group')).toBeInTheDocument();
+
+    // remove first user
+    const removeUser = getAllByTestId('deleteIcon');
+    UserEvent.click(removeUser[0]);
+
+    // click on SAVE
+    const saveButton = screen.getByText('Save');
+    UserEvent.click(saveButton);
+    await wait();
```

Comment:
should we write an expect statement after this?

Reply:
Done